File: sort/selectSort.py
#! python3
# selectSort.py-选择排序

def selectSort(arr):
	for i in range(len(arr) - 1):
		min = i
		for j in range(i, len(arr)): # 选一个最小的
			min = j if arr[j] < arr[min] else min
		arr[i], arr[min] = arr[min], arr[i] # 交换元素
	return arr

# 测试
arr = [-2, 21, 3, 23, 10, -3, 0, -1]
arr = selectSort(arr)
print(arr)

File: data_structure_js/selectSort.js
// 选择排序
function selectSort(arr) {
	for (var i = 0; i < arr.length - 1; i++) {
		var min = i;
		for (var j = i + 1; j < arr.length; j++) {
			if (arr[j] < arr[min]) {
				min = j;
			}
		}
		// min 和 i 交换
		var temp = arr[i];
		arr[i] = arr[min];
		arr[min] = temp;
	}
}

// 测试
var N = 6;
var arr = [];
for (var i = 0; i < N; i++) {
	arr.push(Math.floor(Math.random() * N * 2));
}
console.log(arr);
selectSort(arr);
console.log(arr);